importer/helpers/dagbuilder.go

```diff
@@ -43,14 +52,93 @@ func (dbp *DagBuilderParams) New(in <-chan []byte) *DagBuilderHelper {
 		ncb = nilFunc
 	}
 
-	return &DagBuilderHelper{
+	dbh := &DagBuilderHelper{
 		dserv:    dbp.Dagserv,
 		in:       in,
 		maxlinks: dbp.Maxlinks,
 		ncb:      ncb,
+		pipeline: make(chan *dag.Node),
+		Error:    make(chan error),
+	}
+	dbh.init()
+	return dbh
+}
+
+type step func(*dag.Node) error
+
+func startWorkers(proc step, n int, in, out chan *dag.Node, errch chan error) {
+	var wg sync.WaitGroup
+	for i := 0; i < n; i++ {
+		wg.Add(1)
+		go func() {
+			for node := range in {
+				err := proc(node)
+				if err != nil {
+					errch <- err
```

Comment:
shouldn't there be an early return here? otherwise the error is getting silenced and the pipeline continues. pipeline should exit in most cases.